src/Microsoft.Health.Fhir.Core/Configs/ReindexJobConfiguration.cs

```diff
@@ -46,5 +46,11 @@ public class ReindexJobConfiguration
         /// currently fixed at 1
         /// </summary>
         public ushort MaximumNumberOfConcurrentJobsAllowed { get; internal set; } = 1;
+
+        /// <summary>
+        /// Controls the target percentage of how much of the allocated
+        /// data store resources to use
+        /// </summary>
+        public ushort? TargetDataStoreResourcePercentage { get; set; } = null;
```

Comment:
nit: Perhaps add an example in the comment, Values should be between 0 and 100?, Null means no default limit is set.